Composer/packages/client/src/components/ManageService/ManageService.tsx

```diff
@@ -0,0 +1,702 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { useState, useEffect, Fragment } from 'react';
+import { Dialog, DialogType, DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+import { PrimaryButton, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import formatMessage from 'format-message';
+import { Link } from 'office-ui-fabric-react/lib/Link';
+import { Icon } from 'office-ui-fabric-react/lib/Icon';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import { Spinner } from 'office-ui-fabric-react/lib/Spinner';
+import { useRecoilValue } from 'recoil';
+import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+import { SubscriptionClient } from '@azure/arm-subscriptions';
+import { Subscription } from '@azure/arm-subscriptions/esm/models';
+import { TokenCredentials } from '@azure/ms-rest-js';
+import { CognitiveServicesManagementClient } from '@azure/arm-cognitiveservices';
+import { ResourceManagementClient } from '@azure/arm-resources';
+import { ChoiceGroup, IChoiceGroupOption } from 'office-ui-fabric-react/lib/ChoiceGroup';
+import { ProvisionHandoff } from '@bfc/ui-shared';
+import sortBy from 'lodash/sortBy';
+import { NeutralColors } from '@uifabric/fluent-theme';
+
+import { AuthClient } from '../../utils/authClient';
+import { AuthDialog } from '../../components/Auth/AuthDialog';
+import { armScopes } from '../../constants';
+import { getTokenFromCache, isShowAuthDialog, userShouldProvideTokens } from '../../utils/auth';
+import { dispatcherState } from '../../recoilModel/atoms';
+
+type ManageServiceProps = {
+  createService: (
+    tokenCredentials: TokenCredentials,
+    subscriptionId: string,
+    resourceGroupName: string,
+    resourceName: string,
+    region: string
+  ) => Promise<string>;
+  createServiceInBackground?: boolean;
+  handoffInstructions: string;
+  hidden: boolean;
+  learnMore?: string;
+  onDismiss: () => void;
+  onGetKey: (settings: { key: string; region: string }) => void;
+  onNext?: () => void;
+  serviceName: string;
+  regions?: IDropdownOption[];
+  serviceKeyType: string;
+  setVisible: (value: any) => void;
```

Comment:
this is a callback, name it accordingly, onToggleVisibility: (visible: boolean) => void